contracts/validator-manager/ACP99ValidatorManager.sol

```diff
@@ -0,0 +1,60 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+import {IACP99SecurityModule} from "./interfaces/IACP99SecurityModule.sol";
+import {ValidatorManager} from "./ValidatorManager.sol";
+import {IACP99ValidatorManager, ConversionData, ValidatorRegistrationInput} from "./interfaces/IACP99ValidatorManager.sol";
+
+pragma solidity 0.8.25;
+
+abstract contract ACP99ValidatorManager is IACP99ValidatorManager, ValidatorManager {
+    IACP99SecurityModule public securityModule;
+
+    // TODO: calling this should be restricted to...who?
+    function setSecurityModule(IACP99SecurityModule _securityModule) external {
+        securityModule = _securityModule;
+    }
+
+    function initializeValidatorSet(
+        ConversionData calldata conversionData,
+        uint32 messageIndex
+    ) external {
+        _initializeValidatorSet(conversionData, messageIndex);
+    }
+
+    function initializeValidatorRegistration(
+        ValidatorRegistrationInput calldata input,
+        uint64 weight,
+        bytes calldata args
+    ) external returns (bytes32){
+        bytes32 validationID = _initializeValidatorRegistration(input, weight);
+        securityModule.handleInitializeValidatorRegistration(validationID, weight, args);
```

Comment:
I definitely need to go through the ACP again, but I would expect the security module to do some form of validation on the inputs _before_ calling the `_initializeValidatorRegistration`. Or does it not matter since the security module will just revert anyway?

Reply:
It's a little tricky to reverse the order in our current implementation since the security module needs the validationID, which is provided by the manager contract. It will revert if the security module's checks fail, regardless of the ordering. We should definitely keep in mind patterns such as [checks-effects-interactions](https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html) if/when we decide to move forward on this implementation.